modules/ref_fb_module/src/classifier_fb_impl.cpp

```diff
@@ -0,0 +1,374 @@
+    objPtr.addProperty(outputNameProp);
+    objPtr.getOnPropertyValueWrite("OutputName") +=
+        [this](PropertyObjectPtr& obj, PropertyValueEventArgsPtr& args) { propertyChanged(true); };
+
+    readProperties();
+}
+
+void ClassifierFbImpl::propertyChanged(bool configure)
+{
+    std::scoped_lock lock(sync);
+    readProperties();
+    if (configure)
+        this->configure();
+}
+
+void ClassifierFbImpl::readProperties()
+{
+    useExplicitDomain = objPtr.getPropertyValue("UseExplicitDomain");
+    explicitDimension =  objPtr.getPropertyValue("ExplcitDimension");
+    blockSizeMs = objPtr.getPropertyValue("BlockSizeMs");
+    classCount = objPtr.getPropertyValue("ClassCount");
+    outputName = static_cast<std::string>(objPtr.getPropertyValue("OutputName"));
+
+    assert(blockSizeMs > 0);
+    assert(classCount > 0);
+
+}
+
+FunctionBlockTypePtr ClassifierFbImpl::CreateType()
+{
+    return FunctionBlockType("ref_fb_module_classifier", "Classifier", "Signal classifing");
+}
+
+void ClassifierFbImpl::processSignalDescriptorChanged(const DataDescriptorPtr& inputDataDescriptor,
+                                                   const DataDescriptorPtr& inputDomainDataDescriptor)
+{
+    if (inputDataDescriptor.assigned())
+        this->inputDataDescriptor = inputDataDescriptor;
+    if (inputDomainDataDescriptor.assigned())
+        this->inputDomainDataDescriptor = inputDomainDataDescriptor;
+
+    configure();
+}
+
+void ClassifierFbImpl::configure()
+{
+    if (!inputDataDescriptor.assigned() || !inputDomainDataDescriptor.assigned())
+    {
+        LOG_D("Incomplete signal descriptors")
+        return;
+    }
+
+    try
+    {
+        if (inputDataDescriptor.isStructDescriptor() || inputDataDescriptor.getDimensions().getCount() > 0)
+            throw std::runtime_error("Incompatible input value data descriptor");
+
+        inputSampleType = inputDataDescriptor.getSampleType();
+        if (inputSampleType != SampleType::Float64 &&
+            inputSampleType != SampleType::Float32 &&
+            inputSampleType != SampleType::Int8 &&
+            inputSampleType != SampleType::Int16 &&
+            inputSampleType != SampleType::Int32 &&
+            inputSampleType != SampleType::Int64 &&
+            inputSampleType != SampleType::UInt8 &&
+            inputSampleType != SampleType::UInt16 &&
+            inputSampleType != SampleType::UInt32 &&
+            inputSampleType != SampleType::UInt64)
+            throw std::runtime_error("Invalid sample type");
+
+        if (inputDomainDataDescriptor.getSampleType() != SampleType::Int64 && inputDomainDataDescriptor.getSampleType() != SampleType::UInt64)
+        {
+            throw std::runtime_error("Incompatible domain data sample type");
+        }
+
+        const auto domainRule = inputDomainDataDescriptor.getRule();
+        if (domainRule.getType() != DataRuleType::Linear)
+        {
+            throw std::runtime_error("Domain rule type is not Linear");
+        }
```

Comment:
The function block is then limited to accept only synchronous signals (signals with equidistant samples)? If so, the data processing could be modified a bit, since this means that the block size will always be static, and can be calculated from the linear rule.